go/cmd/dolt/commands/checkout.go

```diff
@@ -93,13 +98,21 @@ func Checkout(ctx context.Context, commandStr string, args []string, dEnv *env.D
 					}
 				}
 
-				if !found {
+				if !found && !env.IsValidDoc(name) {
```

Comment:
Also got rid of this !env.IsValidDoc(name). 

Instead of making a new else-if block I refactored/renamed the existing code to make it clear it is for docs and tables. Let me know what you think.
https://github.com/liquidata-inc/dolt/pull/299/files#diff-78081aa9ea2b0558c0f56aad2a236961R86